octodns/provider/cloudflare.py

```diff
@@ -338,10 +365,29 @@ def populate(self, zone, target=False, lenient=False):
             exists = True
             values = defaultdict(lambda: defaultdict(list))
             for record in records:
-                name = zone.hostname_from_fqdn(record['name'])
-                _type = record['type']
-                if _type in self.SUPPORTS:
-                    values[name][record['type']].append(record)
+                if 'targets' in record:
+                    # assumption, targets will always contain 1 target
+                    # API documentation only indicates 'url' as the only target
+                    # if record['targets'][0]['target'] == 'url':
+                    uri = record['targets'][0]['constraint']['value']
+                    uri = '//' + uri if not uri.startswith('http') else uri
+                    parsed_uri = urlsplit(uri)
+                    name = zone.hostname_from_fqdn(parsed_uri.netloc)
+                    _path = parsed_uri.path
+                    _type = 'URLFWD'
+                    # assumption, actions will always contain 1 action
+                    _values = record['actions'][0]['value']
+                    _values['path'] = _path
+                    # no ttl set by pagerule, creating one
+                    _values['ttl'] = 3600
```

Comment:
I don't think most of these need the `_` on the front. It's just used for `type` since that's a python function.
```suggestion
                    path = parsed_uri.path
                    _type = 'URLFWD'
                    # assumption, actions will always contain 1 action
                    values = record['actions'][0]['value']
                    values['path'] = path
                    # no ttl set by pagerule, creating one
                    values['ttl'] = 3600
```

Reply:
Actually, this was to build a faux DNS record to populate. Applying this change causes problems. And I didn't realize it until after I applied.